src/main/java/im/status/keycard/SecureChannel.java

```diff
@@ -245,24 +349,24 @@ private short pairStep1(byte[] apduBuffer) {
    * @return the length of the reply
    */
   private short pairStep2(byte[] apduBuffer) {
-    crypto.sha256.update(pairingSecret, (short) 0, SC_SECRET_LENGTH);
-    crypto.sha256.doFinal(secret, (short) 0, SC_SECRET_LENGTH, secret, (short) 0);
-
+    // Compare client cryptogram to the expected cryptogram (stored in secret buffer)
     if (Util.arrayCompare(apduBuffer, ISO7816.OFFSET_CDATA, secret, (short) 0, SC_SECRET_LENGTH) != 0) {
       preassignedPairingOffset = -1;
       ISOException.throwIt(ISO7816.SW_SECURITY_STATUS_NOT_SATISFIED);
     }
 
+    // Generate random pairing salt & save pairing key. Copy pairing index and pairing salt into response buffer
+    // pairingKey = sha256(pairingSalt, cryptogram)
```

Comment:
Le cruft?

Reply:
This is my attempt to document how the `pairingKey` is created, even though it kind of looks like commented code. If it is not useful documentation, I can remove.

Note, `pairingKey = sha256(pairingSalt, cryptogram)` is not valid code, just pseudo code describing the process

Reply:
Hah! Yeah I misread, this is fine